docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,609 @@
+# How to Set Up an Edit-Build-Test-Debug Loop
+
+This document describes how to set up a development loop for people interested
+in contributing to Swift.
+
+If you are only interested in building the
+toolchain as a one-off, there are a couple of differences:
+1. You can ignore the parts related to Sccache.
+2. You can stop reading after
+   [Building the project for the first time](#building-the-project-for-the-first-time).
+
+## Table of Contents
+
+- [System Requirements](#system-requirements)
+- [Cloning the project](#cloning-the-project)
+  - [Troubleshooting cloning issues](#troubleshooting-cloning-issues)
+- [Installing dependencies](#installing-dependencies)
+  - [macOS](#macOS)
+  - [Ubuntu Linux](#ubuntu-linux)
+- [Building the project for the first time](#building-the-project-for-the-first-time)
+  - [Spot check dependencies](#spot-check-dependencies)
+  - [Understanding the pieces](#understanding-the-pieces)
+  - [The actual build](#the-actual-build)
+  - [Troubleshooting build issues](#troubleshooting-build-issues)
+- [Editing code](#editing-code)
+  - [Setting up your fork](#setting-up-your-fork)
+  - [First time Xcode setup](#first-time-xcode-setup)
+  - [Editing](#editing)
+  - [Incremental builds with Ninja](#incremental-builds-with-ninja)
+  - [Incremental builds with Xcode](#incremental-builds-with-xcode)
+  - [Spot checking an incremental build](#spot-checking-an-incremental-build)
+- [Reproducing an issue](#reproducing-an-issue)
+- [Running tests](#running-tests)
+- [Debugging issues](#debugging-issues)
+  - [Print debugging](#print-debugging)
+  - [Debugging using LLDB](#debugging-using-lldb)
```

Comment:
print debugging before lldb 🤔

Reply:
I figured that a contributor is more likely to be familiar with print debugging compared to using LLDB, so I put it first. 😅

Reply:
🤷‍♂️